mlir/lib/Conversion/VectorToLLVM/ConvertVectorToLLVM.cpp

```diff
@@ -1623,7 +1626,7 @@ void LowerVectorToLLVMPass::runOnOperation() {
   }
 
   // Convert to the LLVM IR dialect.
-  LLVMTypeConverter converter(&getContext());
+  LLVMTypeConverter converter(&getContext(), llvmOptions);
```

Comment:
@flaub It's used here. It's for the type converter of LLVMIR. The option useBarePtrCallConv in llvmOptions determines how memref is converted (keep as a bare pointer or a couple of variables). The default value is false and we need true.

Reply:
Gotcha, thanks for pointing this out :)